Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -59,13 +59,24 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if(sections.size() != COLUMNS)
+    {
+        return false;
+    }
+
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    bool validTime = batteryData.time.isValid();
 
-    batteryData.voltage = sections.at(1).toDouble();
+    bool validDoubleVoltage;
+    bool validDoubleCurrent;
+    batteryData.voltage = sections.at(1).toDouble(&validDoubleVoltage);
+    batteryData.current = sections.at(2).toDouble(&validDoubleCurrent);
 
-    batteryData.current = sections.at(2).toDouble();
+    if(!validDoubleVoltage || !validDoubleCurrent || !validTime)
+    {
+        return false;
+    }
```

Comment:
You can also just `return(validDoubleVoltage && validDoubleCurrent && validTime)`